egg/language_management.py

```diff
@@ -0,0 +1,66 @@
+import os
+import gettext
+
+
+class LanguageManagement(object):
+    def __init__(self, config=None, file=None):
+        lang = config['default_language_code']
+        self.locales_folder = config['locales_folder']
+        if lang is None and 'LANGUAGE' in os.environ:
+            lang = os.environ['LANGUAGE']
+        else:
+            lang = 'en'
+        self.default_language = lang
+
+        self.current_language = lang
+        self.available_languages = {'en': 'english', 'fr': 'french'}
+        self.detailed_languages = {'en': 'en_US', 'fr': 'fr_FR'}
+        self.translater = {}
+
+        if lang is not None and file is not None and self.locales_folder is not None:
+            self.translater[file] = gettext.translation(
+                domain=file, localedir=self.locales_folder,
+                fallback=True, languages=[lang])
+            self.translater[file].install()
+
+    def get_detailed_locale(self, locale):
+        return self.detailed_languages[locale]
+
+    def get_detailed_locale_country(self, locale):
```

Comment:
Type hints can also be used to annotate a function's return type, the syntax would be:
```suggestion
    def get_detailed_locale_country(self, locale) -> str:
```